### PR TITLE
feat(app): add liner progress bar

### PR DESCRIPTION
**Pull request by koji**
# Overview
Add LinerProgress component for Software update.
The animation part will be update in the hi-fi design.

the design
https://www.figma.com/file/AoTLAYuWawlaWItB1umOjr/OT-3-On-Device-Display?node-id=1776%3A30238&t=npwIVDBH9ZRseGMG-0

I named this component since the spinner that we use for loading page/screen might be updated because pages/screens that fetch data will use Skelton so we might have circular progress bar in the future.

https://user-images.githubusercontent.com/474225/207183825-990ba897-723d-4819-a13e-13b22f1b6659.mov

Close RCORE-453

# Changelog
- add liner progress component
- add stories for liner progress component
- add tests for liner progress component

# Review requests
- Run `make -C components dev` and open LinerProgress

# Risk assessment
low

### CODE REVIEW

File: app/src/atoms/LinerProgress/__tests__/LinerProgress.test.tsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    props = {
+      completed: 0,
+    }
+  })
+
+  afterEach(() => {
+    jest.resetAllMocks()
+  })
+
+  it('renders LinerProgress Bar at 0% with', () => {
```
Comment (jerader) [inline app/src/atoms/LinerProgress/__tests__/LinerProgress.test.tsx]:
```suggestion
  it('renders LinerProgress Bar at 0% width', () => {
```

File: app/src/atoms/LinerProgress/__tests__/LinerProgress.test.tsx [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    const [{ getByTestId }] = render(props)
+    const container = getByTestId('LinerProgress_Container')
+    const bar = getByTestId('LinerProgress_Bar')
+    expect(container).toHaveStyle(`background: ${COLORS.white}`)
+    expect(bar).toHaveStyle('width: 0%')
+  })
+
+  it('renders LinerProgress Bar at 50% with', () => {
```
Comment (jerader) [inline app/src/atoms/LinerProgress/__tests__/LinerProgress.test.tsx]:
```suggestion
  it('renders LinerProgress Bar at 50% width', () => {
```

File: app/src/atoms/LinerProgress/__tests__/LinerProgress.test.tsx [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    props.completed = 50
+    const [{ getByTestId }] = render(props)
+    const bar = getByTestId('LinerProgress_Bar')
+    expect(bar).toHaveStyle(`background: ${COLORS.blueEnabled}`)
+    expect(bar).toHaveStyle('width: 50%')
+  })
+
+  it('renders LinerProgress Bar at 100% with', () => {
```
Comment (jerader) [inline app/src/atoms/LinerProgress/__tests__/LinerProgress.test.tsx]:
```suggestion
  it('renders LinerProgress Bar at 100% width', () => {
```

File: app/src/atoms/LinerProgress/__tests__/LinerProgress.test.tsx [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    props.completed = 100
+    const [{ getByTestId }] = render(props)
+    const bar = getByTestId('LinerProgress_Bar')
+    expect(bar).toHaveStyle(`background: ${COLORS.blueEnabled}`)
+    expect(bar).toHaveStyle('width: 100%')
+  })
+
+  it('renders LinerProgress Bar at 50% + red with', () => {
```
Comment (jerader) [inline app/src/atoms/LinerProgress/__tests__/LinerProgress.test.tsx]:
```suggestion
  it('renders LinerProgress Bar at 50% + red width', () => {
```

File: app/src/atoms/LinerProgress/index.tsx [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+  height?: string
+  /** optional liner progress bar color the default color is blueEnabled */
+  color?: string
+  /** optional liner progress bar radius the default is 0 */
+  borderRadius?: string
+  /** optional  */
+  animationStyle?: string
+}
+
+export function LinerProgress({
```
Comment (jerader) [inline app/src/atoms/LinerProgress/index.tsx]: Can we call this something else like maybe `ProgressBar` or `ProgressMeter`?

**Review by jerader — APPROVED** [review]
good job! it looks really cool! I was wondering if we could rename it to something like `ProgressBar` or `ProgressMeter`? What do you think?

File: app/src/atoms/LinerProgress/__tests__/LinerProgress.test.tsx (hunk 1 above)
Comment (koji) [reply app/src/atoms/LinerProgress/__tests__/LinerProgress.test.tsx]: oh thank you for catching this!

**Comment by koji** [comment]
> good job! it looks really cool! I was wondering if we could rename it to something like `ProgressBar` or `ProgressMeter`? What do you think?

Will change the name to `ProgressBar`

File: app/src/atoms/LinerProgress/index.tsx (hunk 5 above)
Comment (koji) [reply app/src/atoms/LinerProgress/index.tsx]: Changed to `ProgressBar`

File: app/src/atoms/ProgressBar/index.tsx [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+  /** optional liner progress bar color the default color is blueEnabled */
+  color?: string
+  /** optional liner progress bar radius the default is 0 */
+  borderRadius?: string
+  /** optional the default is ease-in-out  */
+  animationStyle?: string
+}
+
+export function ProgressBar({
+  completed,
```
Comment (brenthagen) [inline app/src/atoms/ProgressBar/index.tsx]: i think of `completed` as a boolean status, what about something like `percentComplete` to suggest a 0-100 range (and a comment to that effect)

File: app/src/atoms/ProgressBar/index.tsx [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+  const getCurrentProgress = (): string => {
+    const ratio = completed / 100
+    return ratio > 1 ? '100%' : `${String(ratio * 100)}%`
+  }
+
+  const currentProgress = getCurrentProgress()
+
+  React.useEffect(() => {
+    requestAnimationFrame(() => setProgress(currentProgress))
+  }, [currentProgress])
```
Comment (brenthagen) [inline app/src/atoms/ProgressBar/index.tsx]: what's the advantage of a helper and this useEffect/useState pattern? can't we just directly do:
```
  const ratio = completed / 100
  const progress = ratio > 1 ? '100%' : `${String(ratio * 100)}%`
```

Comment (koji) [reply app/src/atoms/ProgressBar/index.tsx:60]: useEffect is to use `requestAnimationFrame`, but yeah without that still ProgressBar is working as expected.
I removed useEffect and useState.

Comment (koji) [reply app/src/atoms/ProgressBar/index.tsx]: useEffect is to use `requestAnimationFrame`. But as you pointed out, directly passing the value works as expected.
I removed both of them.

**Pull request merged**